ipamutils/utils.go

```diff
@@ -48,3 +63,86 @@ func initGranularPredefinedNetworks() []*net.IPNet {
 	}
 	return pl
 }
+
+// InitAddressPools allows to initialize the local and global scope predefined
+// address pools to the desired values. It fails is invalid input is passed
+// or if the predefined pools were already initialized.
+func InitAddressPools(list []*PredefinedPools) error {
+	var (
+		localPools  []*net.IPNet
+		globalPools []*net.IPNet
+		done        bool
+	)
+
+	for _, p := range list {
+		if p == nil {
+			continue
+		}
+		_, b, err := net.ParseCIDR(p.Base)
+		if err != nil {
+			return fmt.Errorf("invalid base pool %q: %v", p.Base, err)
+		}
+		ones, _ := b.Mask.Size()
+		if p.Size <= 0 || p.Size < ones {
+			return fmt.Errorf("invalid pools size: %d", p.Size)
+		}
+		switch p.Scope {
+		case "", "local":
+			localPools = append(localPools, initPools(p.Size, b)...)
+		case "global":
+			globalPools = append(globalPools, initPools(p.Size, b)...)
+		default:
+			return fmt.Errorf("invalid pool scope: %s", p.Scope)
+		}
+	}
+
+	if localPools == nil {
+		localPools = initBroadPredefinedNetworks()
+	}
+	if globalPools == nil {
+		globalPools = initGranularPredefinedNetworks()
+	}
+
+	initNetworksOnce.Do(func() {
+		PredefinedBroadNetworks = localPools
+		PredefinedGranularNetworks = globalPools
+		done = true
+	})
```

Comment:
Yes, that is correct. If the user passes the default pools, even before creating the network controller daemon will call `InitAddressPools()`